### PR TITLE
remove preprocessing of CriteoDataset, and data auto-split for distributed training in one node, move models/fluid/recommendation to models/fluid/PaddleRec

### PR DESCRIPTION
**Pull request by guru4elephant**

### CODE REVIEW

File: fluid/PaddleRec/ctr/reader.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+class Dataset:
```
Comment (jacquesqiao) [inline fluid/PaddleRec/ctr/reader.py:1]: 这部分是不是应该删掉了？

File: fluid/PaddleRec/ctr/reader.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    def infer(self, file_list):
+        return self._reader_creator(file_list, True)
+
+class CriteoDataset:
+    def __init__(self, sparse_feature_dim):
+        self.cont_min_ = [0, -3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]
+        self.cont_max_ = [20, 600, 100, 50, 64000, 500, 100, 50, 500, 10, 10, 10, 50]
+        self.cont_diff_ = [20, 603, 100, 50, 64000, 500, 100, 50, 500, 10, 10, 10, 50]
+        self.hash_dim_ = sparse_feature_dim
+        self.train_idx_ = 41256555
```
Comment (jacquesqiao) [inline fluid/PaddleRec/ctr/reader.py:12]: train_idx_ 是什么含义？

File: fluid/PaddleRec/ctr/reader.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            for file in file_list:
+                with open(file, 'r') as f:
+                    line_idx = 0
+                    for line in f:
+                        line_idx += 1
+                        if is_train and line_idx > self.train_idx_:
+                            continue
+                        elif not is_train and line_idx <= self.train_idx_:
+                            continue
+                        if trainer_id > 0 and line_idx % trainer_id != 0:
```
Comment (jacquesqiao) [inline fluid/PaddleRec/ctr/reader.py]: This place should use 
```python
line_idx % trainer_num != trainer_idx
```
for example,  `6 % 3` and `6 % 2` is both 0.

File: fluid/PaddleRec/ctr/reader.py (hunk 2 above)
Comment (jacquesqiao) [reply fluid/PaddleRec/ctr/reader.py:12]: need comment

Comment (guru4elephant) [reply fluid/PaddleRec/ctr/reader.py:12]: sure.

File: fluid/PaddleRec/ctr/reader.py (hunk 1 above)
Comment (guru4elephant) [reply fluid/PaddleRec/ctr/reader.py:1]: 可以保留，作为父类。

File: fluid/PaddleRec/ctr/reader.py (hunk 3 above)
Comment (guru4elephant) [reply fluid/PaddleRec/ctr/reader.py]: will fix, thx

**Comment by jacquesqiao** [comment]
preprocess.py can be deleted or keep the function to statistics the range for dens_feature.

**Comment by jacquesqiao** [comment]
LGTM！

**Review by jacquesqiao — APPROVED** [review]
LGTM!

**Pull request merged**